src/core/App.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * @author Lewy Blue / https://github.com/looeee
+ *
+ * parameters = {
+ *
+ *  canvas: <canvas> DOM element,
+ *
+ *  scene: new THREE.Scene(),
+ *
+ *  camera: new THREE.Camera( ... )
+ *
+ *  renderer: new THREE.WebGLRenderer( ... )
+ *
+ *  autoResize: <bool>
+ *
+ *  autoRender: <bool> //setting this to false will prevent calling renderer.render automatically; users can call it manually in the onUpate function instead
+ *
+ * }
+ */
+
+function App( parameters ) {
```

Comment:
I would prefer to pass `canvas` instead of `parameters`. I'm not a big fan of how the code ends up when passing parameters.